hamilton/function_modifiers/metadata.py

```diff
@@ -271,3 +277,43 @@ def example_schema() -> pd.DataFrame:
         Then, when drawing the DAG, the schema will be displayed as sub-elements in the node for the DAG (if `display_schema` is selected).
         """
         return SchemaOutput(*fields, target_=target_)
+
+
+class RayRemote(tag):
+
+    def __init__(self, **options: Union[int, Dict[str, int]]):
+        """Initializes RayRemote. See docs for `@ray_remote` for more details."""
+
+        ray_tags = {f"ray_remote.{option}": json.dumps(value) for option, value in options.items()}
+
+        super(RayRemote, self).__init__(
+            **ray_tags
+        )
+
+    def _key_allowed_wrapper(self, key: str) -> bool:
+        return self._key_allowed(key, reserved_forbidden = False)
+
+
+def ray_remote(**kwargs: Union[int, Dict[str, int]]) -> RayRemote:
```

Comment:
should this be `ray_resources` instead? Don't want to confuse the case when a function doesn't have this that people will think it's not being run via ray remote when it very well could be. 🤔

Reply:
+1 for ray_resources.

Reply:
Also we're using lowercase classnames here -- breaking standard procedure but it'll be more consistent with the rest and remove a layer of indirection.

Reply:
ray.remote options include things beyond resources.
A current list of possible options is here:
`
['max_calls', 'max_retries', 'num_cpus', 'num_returns', 'object_store_memory', 'retry_exceptions', '_generator_backpressure_num_objects', 'concurrency_groups', 'lifetime', 'max_concurrency', 'max_restarts', 'max_task_retries', 'max_pending_calls', 'namespace', 'get_if_exists', 'accelerator_type', 'memory', 'name', 'num_gpus', 'placement_group', 'placement_group_bundle_index', 'placement_group_capture_child_tasks', 'resources', 'runtime_env', 'scheduling_strategy', '_metadata']
`

Reply:
RayRemote matches SchemaOutput, which is also standard CamelCase...this is what I was suggested to base upon.

Reply:
Ahh yep, my bad 😆 

And 👍 on `ray_remote_options`